mlflow/openai/api_request_parallel_processor.py

```diff
@@ -100,77 +100,93 @@ def call_api(self, retry_queue: queue.Queue, status_tracker: StatusTracker):
         """
         _logger.debug(f"Request #{self.index} started")
         try:
-            response = self.task.create(**self.request_json, timeout=self.timeout)
-            _logger.debug(f"Request #{self.index} succeeded")
-            status_tracker.complete_task(success=True)
-            self.results.append((self.index, response))
-        except openai.error.RateLimitError as e:
-            _logger.debug(f"Request #{self.index} failed with {e!r}")
-            current_time = time.time()
-            status_tracker.time_of_last_rate_limit_error = current_time
-            status_tracker.increment_num_rate_limit_errors()
-            # check time since first request, fail at 10 minutes
-            if current_time - self.start_time < 600:
-                if current_time - self.last_log_time > 60:
-                    _logger.warning("Retrying for request failed with rate limit.")
-                    self.last_log_time = current_time
-                retry_queue.put_nowait(self)
-            else:
-                _logger.warning("Request failed after retrying for 10 minutes.")
-                status_tracker.complete_task(success=False)
-                status_tracker.error = mlflow.MlflowException(
-                    "Request failed after retrying for 10 minutes."
-                )
-        # Other retryable errors
-        except (
-            openai.error.Timeout,
-            openai.error.APIError,
-            openai.error.APIConnectionError,
-            openai.error.ServiceUnavailableError,
-        ) as e:
-            _logger.debug(f"Request #{self.index} failed with {e!r}")
-            status_tracker.increment_num_api_errors()
-            if self.attempts_left > 0:
-                retry_queue.put_nowait(self)
-            else:
-                status_tracker.complete_task(success=False)
-                status_tracker.error = e
-        # Unretryable errors
-        except openai.error.InvalidRequestError as e:
-            if e.error.code == "content_filter" and e.error.innererror:
-                content_filter_result = e.error.innererror.content_filter_result
-                _logger.debug(
-                    f"Request #{self.index} failed because of content filtering: "
-                    f"{content_filter_result}"
-                )
-                status_tracker.increment_num_api_errors()
-                status_tracker.complete_task(success=False)
-                status_tracker.error = e
+            import openai
+            request_header = {"Authorization": f"Bearer {openai.api_key}"}
+            # use api-key header for Azure deployments
+            if "/deployments" in self.request_url:
+                request_header = {"api-key": f"{openai.api_key}"}
```

Comment:
can we drop this? that would allow us to not have any openai dep right?

Reply:
We have this dependency in `_OAITokenHolder` where openai.api_key is set - I can try to modify that